packages/react/src/store-cache/useRows.ts

```diff
@@ -0,0 +1,27 @@
+import { DatabaseClient, FilterOptions, ScanResult } from "@latticexyz/store-cache";
+import { StoreConfig } from "@latticexyz/store";
+import { useEffect, useState } from "react";
+
+/**
+ * Returns an array of all rows matching the provided filter
+ */
+export function useRows<C extends StoreConfig, T extends keyof C["tables"] & string>(
+  storeCache: DatabaseClient<C>,
+  filter?: FilterOptions<C, T>
+) {
+  const [rows, setRows] = useState<ScanResult<C, T>>([]);
+
+  useEffect(() => {
+    setRows(storeCache.scan(filter));
+
+    const unsubscribe = storeCache.subscribe(() => {
+      // very naive implementation for now, but easier and probably more efficient than
+      // manually looping through the `rows` array for every update event
+      setRows(storeCache.scan(filter));
+    }, filter);
+
+    return unsubscribe;
+  }, []);
```

Comment:
This empty dependency array is the reason things aren't updating when args change.

See `useEntityQuery` for how I work around this to create a stable reference to an arbitrary object:
https://github.com/latticexyz/mud/blob/91b37a4414fb9730be1886bfbdf4352c37b70d1a/packages/react/src/useEntityQuery.ts#L21

So we'd prob want something like
```ts
const stableStoreCache = useMemo(storeCache); // not deep memo since it should be roughly stable anyway and unlikely inner values would change without an entirely different client
const stableFilter = useDeepMemo(filter); // deep memo since this is more like an argument/set of options

useEffect(() => {
  setRows(stableStoreCache.scan(stableFilter));

  const unsubscribe = stableStoreCache.subscribe(() => {
    // very naive implementation for now, but easier and probably more efficient than
    // manually looping through the `rows` array for every update event
    setRows(stableStoreCache.scan(stableFilter));
  }, stableFilter);

  return unsubscribe;
}, [stableStoreCache, stableFilter]);
```
(I don't love the "stable" term here but also don't love "memoized")